mono/metadata/w32handle.c

```diff
@@ -895,14 +902,75 @@ mono_w32handle_wait_one (gpointer handle, guint32 timeout, gboolean alertable)
 	return ret;
 }
 
+static MonoW32Handle*
+mono_w32handle_has_duplicates (MonoW32Handle *handles [ ], gsize nhandles)
+{
+	if (nhandles < 2 || nhandles > MONO_W32HANDLE_MAXIMUM_WAIT_OBJECTS)
+		return NULL;
+
+	MonoW32Handle *sorted [MONO_W32HANDLE_MAXIMUM_WAIT_OBJECTS]; // 64
+	memcpy (sorted, handles, nhandles * sizeof (handles[0]));
+	qsort (sorted, nhandles, sizeof (gpointer), g_direct_equal);
```

Comment:
Just for clarity and consistency, `qsort (sorted, nhandles, sizeof  (sorted[0]), g_direct_equal);`